src/main/java/io/antmedia/AntMediaApplicationAdapter.java

```diff
@@ -1494,7 +1524,49 @@ private void updateAppSettingsBean(AppSettings appSettings, AppSettings newSetti
 		appSettings.setIpFilterEnabled(newSettings.isIpFilterEnabled());
 		appSettings.setJwtControlEnabled(newSettings.isJwtControlEnabled());
 		appSettings.setJwtSecretKey(newSettings.getJwtSecretKey());
-		
+
+
+		appSettings.setS3RecordingEnabled(newSettings.isS3RecordingEnabled());
+
+		if (appSettings.isS3RecordingEnabled()) {
+			appSettings.setS3AccessKey(newSettings.getS3AccessKey());
+			if (appSettings.getS3AccessKey().isEmpty()) {
```

Comment:
Make logging and operation in the same place.

Logging and these settings are in different places in the code block. It was hard to understand the code  and find the issues